simple-wallet/src/lib.rs

```diff
@@ -171,55 +177,32 @@ where
     }
 }
 
-impl<B: Deref, W: Deref> Signer for SimpleWallet<B, W>
+/// Computes the secret key from the temporary id and the seed.
+pub(crate) fn compute_secret_key(seed: [u8; 32], keys_id: [u8; 32]) -> SecretKey {
+    let mut hmac = HmacEngine::<Sha256>::new(&seed);
+    hmac.input(&keys_id);
+    let secret_bytes = Hmac::from_engine(hmac).into_inner();
+    SecretKey::from_slice(&secret_bytes).expect("Secret key is valid")
```

Comment:
I'm not sure it's ok to use `expect` here. I think there are some values which don't generate a proper secret key.